src/uct/ib/base/ib_md.c

```diff
@@ -846,7 +889,17 @@ static ucs_status_t uct_ib_mem_alloc(uct_md_h uct_md, size_t *length_p,
     uct_ib_mem_t *memh;
     size_t length;
 
-    if (!md->config.enable_contig_pages) {
+#if !HAVE_IBV_EXP_DM
```

Comment:
imp:
```
can combine two `if`s
if (
#if !HAVE_IBV_EXP_DM
     (flags & UCT_MD_MEM_FLAG_ON_DEVICE) ||
#else
     /* in case if ON_DEVICE flag is not active - then memory
      * is allocated using contig_pages and contig pages
      * should be enabled */
     (!(flags & UCT_MD_MEM_FLAG_ON_DEVICE) &&
#endif
      !md->config.enable_contig_pages) {
    return UCS_ERR_UNSUPPORTED;
}
```

it will be only single `if` statement instead of two `if`s in case of `!HAVE_IBV_EXP_DM`. =
but this is not fast path, wdyt?